Tribler/Core/Modules/restapi/events_endpoint.py

```diff
@@ -0,0 +1,63 @@
+from Queue import Queue
+import json
+from twisted.web import server, resource
+from Tribler.Core.Modules.restapi.util import convert_db_channel_to_json, convert_db_torrent_to_json
+from Tribler.Core.simpledefs import NTFY_CHANNELCAST, SIGNAL_CHANNEL, SIGNAL_ON_SEARCH_RESULTS, SIGNAL_TORRENT
+
+
+class EventsEndpoint(resource.Resource):
+    """
+    Important events in Tribler are returned over the events endpoint. This connection is held open. Each event is
+    pushed over this endpoint in the form of a JSON dictionary. Each JSON dictionary contains a type field that
+    indicates the type of the event.
+
+    Currently, the following events are implemented:
+    - events_start: An indication that the event socket is opened and that the server is ready to push events.
+    - search_result_channel: This event dictionary contains a search result with a channel that has been found.
+    - search_result_torrent: This event dictionary contains a search result with a torrent that has been found.
+    """
+
+    def __init__(self, session):
+        resource.Resource.__init__(self)
+        self.session = session
+        self.channel_db_handler = self.session.open_dbhandler(NTFY_CHANNELCAST)
+        self.events_request = None
+        self.buffer = Queue()
+
+        self.session.add_observer(self.on_search_results_channels, SIGNAL_CHANNEL, [SIGNAL_ON_SEARCH_RESULTS])
+        self.session.add_observer(self.on_search_results_torrents, SIGNAL_TORRENT, [SIGNAL_ON_SEARCH_RESULTS])
+
+    def write_data(self, message):
+        """
+        Write data over the event socket. If the event socket is not open, add the message to the buffer instead.
+        """
+        if not self.events_request:
```

Comment:
This will make the Queue grow indefinitely. A limit should be set and older events pushed out if the queue by the newer ones.